view/src/components/surplus.js

```diff
@@ -182,27 +205,54 @@ class Surplus extends Component {
    * @param value New value for the state
    */
   handleAsyncChange = (name, value) => {
+    console.error(name);
+    console.error(value);
     this.setState({
       [name]: value,
     });
   };
 
+  /** Sets states to values so that cards can be reloaded and calls function to reload cards */
+  reFetch = () => {
+    this.setState({
+      uiLoading: true,
+      reLoadCards: true,
+      open: false,
+    });
+    this.reFetchSurplus();
+  };
+
   /** Returns the authentication token stored in local storage */
   getAuth = () => {
     authMiddleWare(this.props.history);
     return localStorage.getItem("AuthToken");
   };
 
+  /** Function to refresh the cards in the page without calling window.reload */
+  reFetchSurplus = () => {
+    axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
+    axios
+      .get("/surplus")
+      .then((response) => {
+        this.setState({
+          surplusObjects: response.data,
+          uiLoading: false,
+          reloadCards: false,
+        });
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
This is a good usage of console.error. Altho in the "real world" you'd probably want a way to surface it to the user (e.g. a toast or an error page). That's probably out of scope here, but maybe could be added to the list of future improvements.

Reply:
I built an alert component.... I'll make sure to add calls to it here!